cardano-node-chairman/src/Chairman/Process.hs

```diff
@@ -128,6 +132,22 @@ waitForProcess :: HasCallStack
 waitForProcess hProcess = GHC.withFrozenCallStack $ do
   H.evalM . liftIO $ catch (fmap Just (IO.waitForProcess hProcess)) $ \(_ :: AsyncCancelled) -> return Nothing
 
+waitSecondsForProcess :: HasCallStack
+  => Int
```

Comment:
Would it make sense to put the initial `waitSecondsForProcess` in a `where` clause here? I'm not sure what the rational is for keeping them separate.

Reply:
In this case I put the `IO` related functions in a separate module from the `Integration` functions so I don't muddle them up and I can give them the same name because they essentially do the same thing.

All the `IO` and `Integration` functions of the same name should do the same thing, the only difference being that the `Integration` functions additionally add annotations, have callstack handling and ensure any exceptions thrown are seen by `hedgehog`.